openevsewifi/__init__.py

```diff
@@ -28,21 +29,80 @@
 colors = ['off', 'red', 'green', 'yellow', 'blue', 'violet', 'teal', 'white']
 
 
+class BadChecksum(Exception):
+    pass
+
+
+class BadResponse(Exception):
+    pass
+
+
+def parse_checksum(s):
+    """
+    If there is a '^' in given string s, this checks that the xor of utf8 bytes
+    before the '^' equal the hex value specified after '^'.  It returns the
+    string before the '^' on success, and throws an BadChecksum exception on error.
+
+    If there is no '^' in the string, the string is returned.
+    """
+    spl = s.rsplit('^', 1)
+    if len(spl) == 1:
+        return s
+    try:
+        check = int(spl[1], 16)
+    except:
```

Comment:
Thanks for adding the exceptions! You still need to change this except clause here though.  As it is, this would raise a BadChecksum exception for _any_ exception, but there should be specific exceptions raised by the previous line that indicate a bad checksum, right?  If this isn't something you're particularly familiar with, [this](Thanks for adding the exceptions! You still need to change this except clause here though.  As it is, this would raise a BadChecksum exception for _any_ exception, but there should be specific exceptions raised by the previous line that indicate a bad checksum, right?  ) article goes a bit more in depth, but let me know if it still leaves you with questions.

Reply:
fixed.